motile/costs/disappear.py

```diff
@@ -30,9 +31,12 @@ def __init__(self, constant: float, ignore_attribute: str | None = None) -> None
 
     def apply(self, solver: Solver) -> None:
         disappear_indicators = solver.get_variables(NodeDisappear)
+        G = solver.graph
 
         for node, index in disappear_indicators.items():
             if self.ignore_attribute is not None:
-                if solver.graph.nodes[node].get(self.ignore_attribute, False):
+                if G.nodes[node].get(self.ignore_attribute, False):
                     continue
+            if G.nodes[node][G.frame_attribute] == G.get_frames()[1]:
```

Comment:
The end frame is actually the last present time frame + 1 (to reflect a slice-like range, see [here](https://github.com/funkelab/motile/blob/825a46198814697623b7287fa43e7f9c163565a9/motile/track_graph.py#L270)). So, if a graph has nodes in times 0, 1, 2, 3, 4, G.get_frames() returns (0, 5). Given that, I think this check is wrong?

Reply:
Good catch. Fixed it and added a test.

On the way,  I also changed the defaults for `TrackGraph.t_begin` and `TrackGraph.t_end` to 0 instead of `None`, that makes for an easier method signature and less overhead for checking for `None`s.